src/collective/ifttt/actions/ifttt.py

```diff
@@ -88,25 +91,35 @@ class IftttTriggerActionExecutor(object):
     """
     timeout = 120
 
-    def __init__(self, element):
+    def __init__(self, element, context, event):
         self.element = element
+        self.context = context
+        self.event = event
 
     def __call__(self, *args, **kwargs):
-        r = self.element.request
-        ifttt_name = self.element.ifttt_name
+        Title = self.element.Title
+        URL = self.element.URL
         payload_option = self.element.payload_option
-        payload = self.element.payload
+        # TODO explore context and event; element is defined with Ifttt Trigger form
+        # TODO define ifttt_url with secret key
+        ifttt_name = self.event.title
+        secret_key = self.event.secret_key
+        ifttt_url = 'https://maker.ifttt.com/trigger/' + ifttt_name + '/with/key/' + secret_key
+        # TODO define payload value
+        payload = {'value1': Title, 'value2': URL, 'value3': ''}
+        if payload_option == PAYLOAD_DESCRIPTION:
+            payload['value3'] = self.event.description
+        elif payload_option == PAYLOAD_USERNAME:
+            payload['value3'] = api.user.get_users()
+        elif payload_option == PAYLOAD_START:
+            payload['value3'] = self.context.start
+        logger.info('Calling Post request to Ifttt')
         try:
-            EXECUTOR.submit(
-                # Post HTTP request
-                r.post,
-                ifttt_name,
-                payload_option,
-                data=payload,
-                timeout=self.timeout
-            )
-        except TypeError:
-            logger.exception('Error calling Ifttt Trigger:')
+            # Post HTTP request
+            requests.post(ifttt_url, data=payload, timeout=self.timeout)
+            logger.info('Successful request call to Ifttt')
```

Comment:
In addition to logging, you may also want to show these to the user using Plone notification messages. Again available through plone.api.